### DISCUSSION TITLE
Shelly TRV (thermostat over valve) - no change in valve position when opening window

### DISCUSSION THREAD

**Discussion by nurunet** [jmcollin78/versatile_thermostat#340] [Q&A]
Hi, I have bought a Shelly TRV because it was recommended here for use with VT. I set it up as a _Thermostat over Valve_. I also created an automation to push the temperature value of an external sensor to the thermostat.

Heating seems to work well, I just have some oscillation above the set heat point - I guess I still have to tweak the TPI settings.

However, I realized I didn't hear the valve move when opening the window. I checked, and - as expected - the Versatile Thermostat UI card shows the open window with little to no delay. The detail page of the VT entity in Home Assistant also shows the valve position to be 0%.

The details of the Shelly TRV entity however show the valve to be still open (at x%), and so does the web UI of the Shelly TRV itself.

Confusingly, the valve then normally moves to 0% when I **close** the window again - at least for a while.

Is this behavior known? Did I misconfigure something? Should I open an issue?

**Comment by jmcollin78** [comment]
Hello @nurunet ,

Many thanks for your report. Yes controling directly the valve is a great feature.

> Heating seems to work well, I just have some oscillation above the set heat point - I guess I still have to tweak the TPI settings.

Yes, you should certainly reduce the coef_int. Do it slowly and wait for stabilization before trying to adjust another time.

I want to warn you about this improvement which have been reported: https://github.com/jmcollin78/versatile_thermostat/issues/338

This is in French but it is possible that the battery of your TRV goes down rapidly. Please check that and wait for the #338  fix if you are in the same case (I put some idea as workaround).

> Is this behavior known? Did I misconfigure something? Should I open an issue?

Not this is not known and look like a bug. I'm very surprise of this because I run automated tests for this case. So I will transform this discussion into an issue.

**Reply by nurunet** [reply]
By the way:  I haven't yet noticed the issue the other user mentioned in #338. I thought I'd have noticed such frequent motor movements... My battery is also still at 100%, but then again the Shelly is pretty new.
